sdk/digitaltwins/digitaltwins/review/digitaltwins.api.md

```diff
@@ -0,0 +1,354 @@
+    publishComponentTelemetry(digitalTwinId: string, componentPath: string, payload: string, messageId?: string, options?: OperationOptions): Promise<RestResponse>;
+    publishTelemetry(digitalTwinId: string, payload: any, messageId?: string, options?: OperationOptions): Promise<RestResponse>;
+    queryTwins(query?: string): PagedAsyncIterableIterator<any, QueryQueryTwinsResponse>;
+    updateComponent(digitalTwinId: string, componentPath: string, componentPatch?: any[], etag?: string, options?: OperationOptions): Promise<DigitalTwinsUpdateComponentResponse>;
+    updateDigitalTwin(digitalTwinId: string, twinPatch: any, etag?: string, options?: OperationOptions): Promise<DigitalTwinsUpdateResponse>;
+    updateRelationship(digitalTwinId: string, relationshipId: string, relationshipPatch: any[], etag?: string, options?: OperationOptions): Promise<DigitalTwinsUpdateRelationshipResponse>;
+    upsertDigitalTwin(digitalTwinId: string, digitalTwinJson: string, options?: OperationOptions): Promise<DigitalTwinsAddResponse>;
+    upsertEventRoute(eventRouteId: string, endpointId: string, filter?: string, options?: OperationOptions): Promise<RestResponse>;
+    upsertRelationship(digitalTwinId: string, relationshipId: string, relationship: any, options?: OperationOptions): Promise<DigitalTwinsAddRelationshipResponse>;
+}
+
+// @public
+export interface DigitalTwinsDeleteRelationshipOptionalParams extends coreHttp.OperationOptions {
+    ifMatch?: string;
+}
+
+// @public
+export interface DigitalTwinsGetByIdHeaders {
+    etag?: string;
+}
+
+// @public
+export type DigitalTwinsGetByIdResponse = DigitalTwinsGetByIdHeaders & {
+    body: any;
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: any;
+        parsedHeaders: DigitalTwinsGetByIdHeaders;
+    };
+};
+
+// @public
+export interface DigitalTwinsGetComponentHeaders {
+    etag?: string;
+}
+
+// @public
+export type DigitalTwinsGetComponentResponse = DigitalTwinsGetComponentHeaders & {
+    body: any;
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: any;
+        parsedHeaders: DigitalTwinsGetComponentHeaders;
+    };
+};
+
+// @public
+export type DigitalTwinsListIncomingRelationshipsResponse = IncomingRelationshipCollection & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: IncomingRelationshipCollection;
+    };
+};
+
+// @public
+export interface DigitalTwinsListRelationshipsOptionalParams extends coreHttp.OperationOptions {
+    relationshipName?: string;
+}
+
+// @public
+export type DigitalTwinsListRelationshipsResponse = RelationshipCollection & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: RelationshipCollection;
+    };
+};
+
+// @public
+export interface DigitalTwinsSendComponentTelemetryOptionalParams extends coreHttp.OperationOptions {
+    dtTimestamp?: string;
```

Comment:
feels weird to have `dt` as a prefix to `Timestamp` is it somehow special to digital twins? Or is it just a normal epoch stamp?

Reply:
No, it is not special. This is the definition from the generated code:
  /**
   * An RFC 3339 timestamp that identifies the time the telemetry was measured.
   */
  dtTimestamp?: string;

But because of generated we have to go with it for now.

---
In reply to: [470197295](https://github.com/Azure/azure-sdk-for-js/pull/10560#discussion_r470197295) [](ancestors = 470197295)

Reply:
If the problem is "it has a bad name in swagger" the solution is to use `x-ms-client-name` in a swagger transform to make the generated name better.

Reply:
+1 for `timestamp` from my perspective